primitives/arithmetic/src/per_things.rs

```diff
@@ -200,6 +258,31 @@ macro_rules! implement_per_thing {
 				// This will always fit into $type.
 				Self::from_parts(parts as $type)
 			}
+
+			/// Saturating exponentiation. Computes `self.pow(exp)`, saturating at the numeric
+			/// bounds instead of overflowing. This operation is lossy.
+			fn saturating_pow(self, exp: usize) -> Self {
+				if self.is_zero() || self.is_one() {
+					self
+				} else {
+					let p = <$name as PerThing>::Upper::from(self.deconstruct());
+					let q = <$name as PerThing>::Upper::from(Self::ACCURACY);
+					let mut s = Self::one();
+					for _ in 0..exp {
```

Comment:
I guess this could be improved by keeping intermediary result, like `x^10 == ((x^2)^2)^2*x^2` but anyway seems ok